py/desitarget/gaiamatch.py

```diff
@@ -222,7 +237,8 @@ def sub_gaia_edr3(filename, objs=None):
 
     # ADM read the gaia sweep.
     cols = ["PARALLAX", "PARALLAX_IVAR",
-            "PMRA", "PMRA_IVAR", "PMDEC", "PMDEC_IVAR"]
+            "PMRA", "PMRA_IVAR", "PMDEC", "PMDEC_IVAR",
+            "GAIA_ASTROMETRIC_PARAMS_SOLVED"]
```

Comment:
I think this will fail? The parameter is called `EDR3_ASTROMETRIC_PARAMS_SOLVED` in the Gaia EDR3 sweeps files, so for this additional parameter you need to swap out `GAIA` for `EDR3` rather than just prepending `EDR3`?

Reply:
Good catch, thanks.

Reply:
This is almost correct now, but not quite. You need to turn [these columns](https://github.com/desihub/desitarget/blob/5b2a280bb303eef17d49ec748e631047c4e8fb3a/py/desitarget/gaiamatch.py#L63-L66):
```
    ('GAIA_ASTROMETRIC_SIGMA5D_MAX', '>f4'), ('GAIA_ASTROMETRIC_PARAMS_SOLVED', '>i1'),
    ('PARALLAX', '>f4'), ('PARALLAX_IVAR', '>f4'),
    ('PMRA', '>f4'), ('PMRA_IVAR', '>f4'),
    ('PMDEC', '>f4'), ('PMDEC_IVAR', '>f4')
```
Into [these columns](https://github.com/desihub/desitarget/blob/5b2a280bb303eef17d49ec748e631047c4e8fb3a/py/desitarget/gaiamatch.py#L93-L98):
```
    ('EDR3_ASTROMETRIC_SIGMA5D_MAX', '>f4'), ('EDR3_ASTROMETRIC_PARAMS_SOLVED', '>i1'), 
    ('EDR3_PARALLAX', '>f4'), ('EDR3_PARALLAX_IVAR', '>f4'),
    ('EDR3_PMRA', '>f4'), ('EDR3_PMRA_IVAR', '>f4'),
    ('EDR3_PMDEC', '>f4'), ('EDR3_PMDEC_IVAR', '>f4')
```
So the `PMRA` and `PARALLAX` columns need an `EDR3` prepended but the `GAIA_ASTROMETRIC_SIGMA5D_MAX` and `GAIA_ASTROMETRIC_PARAMS_SOLVED` columns need `GAIA` switched to `EDR3`. So, some code logic like:

```
gaiacols = [col.replace("GAIA", "EDR3") if "GAIA" in col else "EDR3_{}".format(col) for col in cols]
```
should work.

Sorry this is so tricky. I inherited the Gaia DR2 data model from the Legacy Surveys catalogs.

Reply:
OK, thanks. Sorry I'm clearly not following very well how the columns get copied. Will make those changes.

Also I thought the discussion with Boris above ended on GAIA_ASTROMETRIC_SIGMA5D_MAX being left as DR2. For consistency it seems better to me to change it, so I will follow your suggestion.